docs/modules/ROOT/pages/how-to-guides/proc_prefetching-dependencies-to-support-hermetic-build.adoc

```diff
@@ -0,0 +1,195 @@
++
+NOTE: This script has no runtime dependency other than `pip`.
+
+.Procedures
+For each component that you want to build hermetically:
+
+. Navigate to your component's source code.
+. Open the *.tekton* directory and locate the *.yaml* files related to the *pull-request* and *push* processes.
+. *Set up dependency prefetching with Cachi2:* In both the `.yaml` files, add the following prefetch-input pipeline parameter.
+
++
+[source,yaml]
+----
+param:
+    -  name: prefetch-input
+       value: <package-manager-lock-file> <1>
+----
+<1> If the package manager lock file is located in the root directory of your project, enter `pip`. Otherwise, if the package manager lock file is in a subdirectory, provide the path to the lock file in the JSON format. For example,
+{"type": "{<package-manager>}", "path": "path-to-the-directory-which-contains-the-lockfile}"}
+
+. Create a file (for example, pyproject.toml) with metadata (name and version) and place it in the root of your repository.
+
++
+[source,metadata]
+----
+[metadata]
+name = "my_package"
+version = "0.1.0"
+----
+
++
+NOTE: Instead of a *pyproject.toml* file, you can also create a *setup.py* or *setup.cfg* file. For information on the metadata of these files, see link:https://github.com/containerbuildsystem/cachi2/blob/main/docs/pip.md#project-metadata[Project Metadata].
+
+. Generate a fully resolved requirements.txt file by using the following command:
+
++
+[source,command]
+----
+$ pip-compile pyproject.toml --generate-hashes
+----
+
++
+NOTE: The command mentioned assumes that you have defined project dependencies in *pyproject.toml*. However, if you have defined the project dependencies in either the *setup.py* or *setup.cfg*, ensure to update the command accordingly.
+
+. In the root of your repository create a file, *requirements-build.in*.
+
+. Copy the build system requirements from the *pyproject.toml* file to the *requirements-build.in* file.
+
++
+*For example:*
++
+image::build-requirements.png[Build requirements, role="image"]
+
+. Run the `pip_find_builddeps.py` script and `pip-compile` the outputs by using the following command:
+
++
+[source,command]
+----
+$ pip_find_builddeps.py requirements.txt \
+--append \
+--only-write-on-update \
+-o requirements-build.in
+----
+
+. Use the `pip-compile` command to convert the *requirements-build.in* file in to the *requirements-build.txt* file by using the following command:
+
++
+[source,command]
+----
+$ pip-compile requirements-build.in --allow-unsafe --generate-hashes
+----
+
+. Add the *requirement-build.txt* file to your project. It does not require any changes to your build process. 
+
++
+NOTE: `pip` automatically installs the build dependencies when needed for explicit installation. The purpose of the *requirement-build.txt* file is to enable Cachi2 to fetch the build dependencies and provide them to `pip` for offline installation in a network-isolated environment.
```

Comment:
I don't think that this is always true. Sometimes changes are needed to the build process so that certain build-time dependencies can be built first before they are needed.

Reply:
@arewm Your thoughts on this: 
NOTE: `pip` automatically installs the necessary build dependencies when explicitly installing them. However, it's important to note that in certain scenarios, modifications to the build process may be required to ensure that specific build-time dependencies are built before they are actually required. In such cases, you can use the `requirements.txt` file to specify the build dependencies that must be installed. The purpose of the *requirement-build.txt* file is to enable Cachi2 to fetch the build dependencies and provide them to `pip` for offline installation in a network-isolated environment.

Reply:
I think it's true enough to leave out the extra complexity. Pip - at least up to date versions of pip - does always install build dependencies automatically.

There are of course situations, especially if setuptools and pip in your base image are very outdated, where you can run into baffling bugs in build dependency handling

Reply:
ack, thanks. I am fine leaving out for now then. Can we represent this in https://github.com/containerbuildsystem/cachi2/blob/main/docs/pip.md#troubleshooting so that users may be able to discover it without polluting this documentation?

Reply:
It's meant to be https://github.com/containerbuildsystem/cachi2/blob/main/docs/pip.md#miscellaneous-errors-while-building-from-source

Reply:
Fair enough -- it isn't directly what I was thinking of for a solution suggestion, but we can always update that. I think that this thread can be resolved without further actions. Any objections?

Reply:
I am marking this thread as resolved without any changes to the content.